scripts/submodules/versions.mk

```diff
@@ -0,0 +1,123 @@
+				    test -z "$(BUILD_REVISION)" || $(MAKE) test-$(1); \
+			        touch .validate-versions-failure; \
+				fi; \
+	        elif test "x$($(2)_BRANCH)" != "x$(NEEDED_$(2)_BRANCH)" ; then \
+				if test x$$(RESET_VERSIONS) != "x"; then \
+					test -z "$(BUILD_REVISION)" || $(MAKE) test-$(1); \
+					$(MAKE) reset-$(1) || exit 1; \
+				else \
+				    echo "Your $(1) branch is out of date, please run 'make reset-$(1)' (found $($(2)_BRANCH), expected $(NEEDED_$(2)_BRANCH))"; \
+			        touch .validate-versions-failure; \
+				fi; \
+	       fi; \
+	    fi; \
+	fi
+
+test-$(1)::
+	@echo $(1)
+	@echo "   REPOSITORY_$(2)=$(REPOSITORY_$(2))"
+	@echo "   DIRECTORY_$(2)=$(DIRECTORY_$(2))"
+	@echo "   MODULE_$(2)=$(MODULE_$(2))"
+	@echo "   NEEDED_$(2)_VERSION=$(NEEDED_$(2)_VERSION)"
+	@echo "   $(2)_VERSION=$($(2)_VERSION)"
+	@echo "   $(2)_BRANCH_AND_REMOTE=$($(2)_BRANCH_AND_REMOTE)"
+	@echo "   NEEDED_$(2)_BRANCH=$(NEEDED_$(2)_BRANCH)"
+	@echo "   NEEDED_$(2)_REMOTE=$(NEEDED_$(2)_REMOTE)"
+	@echo "   $(2)_BRANCH=$($(2)_BRANCH)"
+	@echo "   $(2)_PATH=$($(2)_PATH) => $(abspath $($(2)_PATH))"
+
+reset-$(1)::
+	@if test -d $($(2)_PATH); then \
+		if ! (cd $($(2)_PATH) && git show $(NEEDED_$(2)_VERSION) >/dev/null 2>&1 && git log -1 $(NEEDED_$(2)_REMOTE) >/dev/null 2>&1) ; then \
+			echo "*** git fetch `basename $$($(2)_PATH)`" && (cd $($(2)_PATH) && git fetch); \
+		fi;  \
+	else \
+		echo "*** git clone $(MODULE_$(2)) --recursive $(DIRECTORY_$(2))" && (cd `dirname $($(2)_PATH)` && git clone $(MODULE_$(2)) --recursive $(DIRECTORY_$(2)) || exit 1 ); \
+	fi
+	@if test x$$(IGNORE_$(2)_VERSION) = "x"; then \
+		echo "*** [$(1)] git checkout -f" $(NEEDED_$(2)_BRANCH) && (cd $($(2)_PATH) ; git checkout -f $(NEEDED_$(2)_BRANCH) || git checkout -f -b $($(2)_BRANCH_AND_REMOTE)); \
+		echo "*** [$(1)] git reset --hard $(NEEDED_$(2)_VERSION)" && (cd $($(2)_PATH) && git reset --hard $(NEEDED_$(2)_VERSION)); \
+	fi
+	@echo "*** [$(1)] git submodule update --init --recursive" && (cd $($(2)_PATH) && git submodule update --init --recursive)
+
+print-$(1)::
+	@printf "*** %-16s %-45s %s (%s)\n" "$(DIRECTORY_$(2))" "$(MODULE_$(2))" "$(NEEDED_$(2)_VERSION)" "$(NEEDED_$(2)_BRANCH)"
+
+.PHONY: validate-$(1) reset-$(1) print-$(1)
+
+reset-versions:: reset-$(1)
+validate-versions:: validate-$(1)
+print-versions:: print-$(1)
+
+endef
+
+reset-versions::
+
+validate-versions::
+	@if test -e .validate-versions-failure; then  \
+		rm .validate-versions-failure; \
+		echo One or more modules needs update;  \
+		exit 1; \
+	else \
+		echo All dependent modules up to date;  \
+	fi
+
+reset:
+	@$(MAKE) validate-versions RESET_VERSIONS=1
+
+__bump-version-%:
+	@if [ "$(REV)" = "" ]; then echo "Usage: make bump-version-$* REV=<ref>"; exit 1; fi
+	python $(SCRIPT) set-rev $* $(REV)
+	@if [ "$(COMMIT)" = "1" ]; then echo "[acceptance-tests] Bump $* to pick up $(REV)." | git commit -F - $(CONFIG); fi
```

Comment:
the `[acceptance-tests]` prefix doesn't make sense in the generalized version. Same on L118 and L123.